core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -54,11 +57,17 @@
  * necessary.
  */
 class DelayedStream implements ClientStream {
-  private final ClientStreamListener listener;
 
-  // Volatile to be readable without synchronization in the fast path.
-  // Writes are also done within synchronized(this).
-  private volatile ClientStream realStream;
+  // set to non null once both listener and unstartedStream are valid.  After this point it is safe
+  // to call methods on startedStream.  Note: this may be true even after the delayed stream is
+  // cancelled.  This should be okay.
+  private volatile ClientStream startedStream;
+  @GuardedBy("this")
+  private ClientStreamListener listener;
+  @GuardedBy("this")
+  private volatile ClientStream unstartedStream;
```

Comment:
This doesn't look like it needs to be volatile. All accesses are within synchronized blocks.